lib/plugins/place_block.js

```diff
@@ -4,13 +4,14 @@ module.exports = inject
 
 function inject (bot) {
   async function placeBlockWithOptions (referenceBlock, faceVector, options) {
-    bot.swingArm()
-    const pos = await bot._genericPlace(referenceBlock, faceVector, options)
+    const dest = referenceBlock.position.plus(faceVector)
+    let oldBlock = bot.blockAt(dest)
+    await bot._genericPlace(referenceBlock, faceVector, options)
 
-    const dest = pos.plus(faceVector)
-    const eventName = `blockUpdate:${dest}`
-
-    const [oldBlock, newBlock] = await onceWithCleanup(bot, eventName, { timeout: 5000 })
+    let newBlock = bot.blockAt(dest)
+    if (oldBlock.type === newBlock.type) {
+      [oldBlock, newBlock] = await onceWithCleanup(bot, `blockUpdate:${dest}`, { timeout: 5000 })
+    }
```

Comment:
This approach is overly complicated, switch to @u9g's suggestion of just awaiting both `_genericPlace` and `blockUpdate` together if possible